File: programers_Lv2/행렬의 곱셈.py
def solution(arr1, arr2):
    answer = []

    new_arr2 = []
    for i in range(len(arr2[0])):
        temp = []
        for v in arr2:
            temp.append(v[i])
        new_arr2.append(temp)

    def func(arr1, arr2):
        rst = 0
        for i in range(len(arr1)):
            rst += arr1[i] * arr2[i]
        return rst

    for v in arr1:
        temp = []
        for w in new_arr2:
            temp.append(func(v, w))
        answer.append(temp)

    return answer


print(
    solution([[1, 4], [3, 2], [4, 1]], [[3, 3], [3, 3]])
)  # [[15, 15], [15, 15], [15, 15]]

File: programers_Lv2/행렬의 곱셈.js
function solution(arr1, arr2) {
  var answer = [];

  // newArr2 = [];
  // arr2[0].forEach((_, i) => {
  //   const temp = [];
  //   arr2.forEach((v, j) => {
  //     temp.push(v[i]);
  //   });
  //   newArr2.push(temp);
  // });

  newArr2 = arr2[0].reduce((acc, _, idx) => {
    let rst = [];
    arr2.forEach((v) => {
      rst.push(v[idx]);
    });
    acc = [...acc, rst];
    return acc;
  }, []);

  const func = (arr1, arr2) => {
    let rst = 0;
    arr1.forEach((_, i) => {
      rst += arr1[i] * arr2[i];
    });
    return rst;
  };

  arr1.forEach((v) => {
    const rst = [];
    newArr2.forEach((w) => rst.push(func(v, w)));
    answer.push(rst);
  });

  return answer;
}

console.log(
  solution(
    [
      [1, 4],
      [3, 2],
      [4, 1],
    ],
    [
      [3, 3],
      [3, 3],
    ]
  )
); // [[15, 15], [15, 15], [15, 15]]
